apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts

```diff
@@ -48,6 +49,11 @@ export async function generateDeploymentScripts(context: IActionContext): Promis
     const connectionsData: ConnectionsData = isEmptyString(connectionsJson) ? {} : JSON.parse(connectionsJson);
     const isParameterized = await isConnectionsParameterized(connectionsData);
     const workflowFiles = getWorkflowFilePaths(projectPath);
+    await ConvertToWorkspace(context);
+    if (context.telemetry.properties.isWorkspace !== 'true') {
```

Comment:
NIT: While this works, relying on telemetry for control flow decisions could make the code harder to maintain and test and not one of the best practices. Consider having ConvertToWorkspace return a boolean to explicitly indicate success/failure, which would make the behavior more predictable and easier to verify.

Reply:
Also you can even do it in a try/catch statement if something goes wrong and we can log telemetry